### ISSUE TITLE
How to train my own model with katib on kubeflow

### ISSUE THREAD

**Issue by RoterDL** [kubeflow/katib#718]
I want to use my own data to train on kubeflow, but I know less about K8S. Where should I put my data set so that the program can read it? Thanks!

**Comment by gaocegege** [comment]
Hi 

Where are your data? If your data is in HDFS, you can read it in the code. If your data is in the disk, you need to create a PV and claim a PVC. In katib, you can mount the PVC. Then you can use the data with file IO.

**Comment by RoterDL** [comment]
Thank you for your reply.I confirm that I have mounted the PVC.  This is my yaml file.
`apiVersion: "kubeflow.org/v1alpha2"
kind: Experiment
metadata:T
  namespace: kubeflow
  name: random-experiment
spec:
  parallelTrialCount: 3
  maxTrialCount: 12
  maxFailedTrialCount: 3
  objective:
    type: maximize
    goal: 0.99
    objectiveMetricName: accuracy
  algorithm:
    algorithmName: random
  trialTemplate:
    retain: true
    goTemplate:
        rawTemplate: |-
          apiVersion: "kubeflow.org/v1"
          kind: PyTorchJob
          metadata:
            name: {{.Trial}}
            namespace: {{.NameSpace}}
          spec:
           pytorchReplicaSpecs:
            Master:
              replicas: 1
              restartPolicy: OnFailure
              template:
                spec:
                  containers:
                    - name: pytorch
                      image: nydsjhub.com/kubeflow/pytorch:v1.0
                      imagePullPolicy: Always
                      volumeMounts:
                          - name: tensorflow-py
                            mountPath: "/mnt/nydsjsmb/kubelfow"
                    volumes:
                      - name:  tensorflow-py
                        hostPath:
                          path: /mnt/nydsjsmb/kubelfow
                      imagePullPolicy: Always
                      command:
                        - "python"
                        - "/mnt/nydsjsmb/kubelfow/pythoch.py"
                        {{- with .HyperParameters}}
                        {{- range .}}
                        - "{{.Name}}={{.Value}}"
                        {{- end}}
                        {{- end}}
                      #volumeMounts:
                      #  - name: tensorflow-py
                      #    mountPath: "/mnt/nydsjsmb/kubelfow"
                    #volumes:
                    #  - name:  tensorflow-py
                    #    hostPath:
                    #      path: /mnt/nydsjsmb/kubelfow
            Worker:
              replicas: 2
              restartPolicy: OnFailure
              template:
                spec:
                  cont`
After running, the error is as follows:
`Invalid spec.trialTemplate: error converting YAML to JSON: yaml: line 19: did not find expected '-' indicator. `
Is my yaml file configuration problem?

**Comment by cleveryg** [comment]
apiVersion: "kubeflow.org/v1alpha2"
kind: Experiment
metadata:
  namespace: kubeflow
  name: luote-py
spec:
  parallelTrialCount: 3
  maxTrialCount: 12
  maxFailedTrialCount: 3
  objective:
    type: maximize
    goal: 0.99
    objectiveMetricName: accuracy
  algorithm:
    algorithmName: random
  trialTemplate:
    retain: true
    goTemplate:
        rawTemplate: |-
          apiVersion: "kubeflow.org/v1"
          kind: PyTorchJob
          metadata:
            name: {{.Trial}}
            namespace: {{.NameSpace}}
          spec:
           pytorchReplicaSpecs:
            Master:
              replicas: 1
              restartPolicy: OnFailure
              template:
                spec:
                  containers:
                    - name: pytorch
                      image: nydsjhub.com/kubeflow/pytorch:v1.0
                      #imagePullPolicy: Always
                      volumeMounts:
                      - name: tensorflow-py
                        mountPath: "/mnt/nydsjsmb/kubelfow"
                    volumes:
                      - name:  tensorflow-py
                        hostPath:
                          path: /mnt/nydsjsmb/kubelfow
                      imagePullPolicy: Always
                      command:
                        - "python"
                        - "/mnt/nydsjsmb/kubelfow/pythoch.py"
                        {{- with .HyperParameters}}
                        {{- range .}}
                        - "{{.Name}}={{.Value}}"
                        {{- end}}
                        {{- end}}
            Worker:
              replicas: 2
              restartPolicy: OnFailure
              template:
                spec:
                  containers:
                    - name: pytorch
                      image: nydsjhub.com/kubeflow/pytorch:v1.0
                      imagePullPolicy: Always
                      #volumeMounts:
                      #  - name: tensorflow-py1
                      #    mountPath: "/mnt/nydsjsmb/kubelfow"
                    #volumes:
                    #  - name:  tensorflow-py1
                    #    hostPath:
                    #      path: /mnt/nydsjsmb/kubelfow
                      command:
                        - "python"
                        - "/mnt/nydsjsmb/kubelfow/pythoch.py"
                        {{- with .HyperParameters}}
                        {{- range .}}
                        - "{{.Name}}={{.Value}}"
                        {{- end}}
                        {{- end}}
  parameters:
    - name: --lr
      parameterType: double
      feasibleSpace:
        min: "0.01"
        max: "0.05"
    - name: --momentum
      parameterType: double
      feasibleSpace:
        min: "0.5"
        max: "0.9"

**Comment by cleveryg** [comment]
apiVersion: "kubeflow.org/v1alpha2"
kind: Experiment
metadata:
  namespace: kubeflow
  name: random-experiment
spec:
  parallelTrialCount: 3
  maxTrialCount: 12
  maxFailedTrialCount: 3
  objective:
    type: maximize
    goal: 0.99
    objectiveMetricName: accuracy
  algorithm:
    algorithmName: random
  trialTemplate:
    retain: true
    goTemplate:
        rawTemplate: |-
          apiVersion: "kubeflow.org/v1"
          kind: PyTorchJob
          metadata:
            name: {{.Trial}}
            namespace: {{.NameSpace}}
          spec:
           pytorchReplicaSpecs:
            Master:
              replicas: 1
              restartPolicy: OnFailure
              template:
                spec:
                  containers:
                    - name: pytorch
                      image: nydsjhub.com/kubeflow/pytorch-dist-mnist-test:v1.0
                      imagePullPolicy: Always
                      command:
                        - "python"
                        - "/var/mnist.py"
                        {{- with .HyperParameters}}
                        {{- range .}}
                        - "{{.Name}}={{.Value}}"
                        {{- end}}
                        {{- end}}
            Worker:
              replicas: 2
              restartPolicy: OnFailure
              template:
                spec:
                  containers:
                    - name: pytorch
                      image: nydsjhub.com/kubeflow/pytorch-dist-mnist-test:v1.0
                      imagePullPolicy: Always
                      command:
                        - "python"
                        - "/var/mnist.py"
                        {{- with .HyperParameters}}
                        {{- range .}}
                        - "{{.Name}}={{.Value}}"
                        {{- end}}
                        {{- end}}
  parameters:
    - name: --lr
      parameterType: double
      feasibleSpace:
        min: "0.01"
        max: "0.05"
    - name: --momentum
      parameterType: double
      feasibleSpace:
        min: "0.5"
        max: "0.9"

**Comment by johnugeorge** [comment]
here is one example that you use volumes in TrialSpec(uses TFJob in this example)

https://github.com/kubeflow/katib/blob/master/examples/v1alpha2/tfjob-example.yaml#L59

**Comment by cleveryg** [comment]
error":"rpc error: code = Unknown desc = Exception calling application: AbortionError(code=StatusCode.UNAVAILABLE, details=\"Connection reset by peer\")"

**Comment by gaocegege** [comment]
/close 

It is stale, but feel free to leave comments here.
